src/main.ts

```diff
@@ -87,13 +100,19 @@ async function run(): Promise<void> {
       .map(file => file.filename)
       .filter(filename => filesToWatch.includes(filename))
     const prCompliant =
-      bodyCheck && titleCheck && branchCheck && filesFlagged.length === 0
+      bodyCheck &&
+      issueLabelErrors.length === 0 &&
+      titleCheck &&
+      branchCheck &&
+      filesFlagged.length === 0
     const shouldClosePr =
       (bodyCheck === false && bodyAutoClose === true) ||
-      (branchCheck === false && protectedBranchAutoClose === true)
+      (branchCheck === false && protectedBranchAutoClose === true) ||
+      (!!issueLabelErrors.length && issueLabelsAutoClose === true)
```

Comment:
any reason not to do .length > 0?

Reply:
No reason, just a stylistic preference on my end. Fewer characters 🧹